### DISCUSSION TITLE
How to only include a certain directory

### DISCUSSION THREAD

**Discussion by KlemensGenetec** [coverlet-coverage/coverlet#1406] [Q&A] [answered]
Hello all!

We have a microservice architecture with multiple services being hosted in a single repository. When we are building a service, we only build and test a certain set of projects. When we are calculating the coverage, it is pulling in a lot of code from other services somehow that are not tested in the service. So the coverage rate seems really bad. 
How can we tell dotnet test to just include the project in a single location? Excluding all projects that shouldn't be there is an option but not very maintainable. 
Is there a better way to do that? I tried with IncludeDirectory but that didnt change the result.

Thanks!

**Comment by Bertk** [comment, marked as answer]
Hi, please use filter options **Exclude** and **Include** ([Advanced Options (Supported via runsettings)](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md#advanced-options-supported-via-runsettings))

Example for `dotnet test` and coverlet.msbuild
```shell
/p:Exclude="[coverlet.core.tests.samples.netstandard]*%2c[coverlet.tests.projectsample]*"
```
